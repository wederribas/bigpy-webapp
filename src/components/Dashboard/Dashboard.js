import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import { Paper, Typography, withStyles } from '@material-ui/core'
import UserReportFeed from '../UserReportFeed/UserReportFeed'
import BouncingLoader from '../BouncingLoader/BouncingLoader'
import ErrorMessage from '../ErrorMessage/ErrorMessage'

const styles = theme => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  loader: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: theme.spacing.unit * 30
  }
})

export const DASHBOARD_QUERY = gql`
  query UsersReportsQuery($companyName: String!) {
    totalCompanyReports(companyName: $companyName)
  }
`

function Loading(props) {
  return (
    <div className={props.classes.loader}>
      <BouncingLoader />
    </div>
  )
}

Loading.propTypes = {
  classes: PropTypes.object.isRequired
}

function CompanyInformationBar(props) {
  return (
    <Paper className={props.classes.root} elevation={1}>
      <Typography data-testid="company-name" variant="h4" component="h2">
        {props.company.name}
      </Typography>
      <Typography data-testid="reports-count" component="p">
        {props.company.reportsCount}
      </Typography>
    </Paper>
  )
}

CompanyInformationBar.propTypes = {
  classes: PropTypes.object.isRequired,
  company: PropTypes.object.isRequired
}

class Dashboard extends Component {
  state = {}
  render() {
    const { classes } = this.props
    const companyName = this.props.match.params.companyName
    return (
      <div>
        <Query query={DASHBOARD_QUERY} variables={{ companyName: companyName }}>
          {({ loading, error, data }) => {
            if (loading) return <Loading classes={classes} />
            if (error) return <ErrorMessage />

            return (
              <div>
                <CompanyInformationBar
                  classes={classes}
                  company={{
                    name: companyName,
                    reportsCount: data.totalCompanyReports
                  }}
                />
                <UserReportFeed companyName={companyName} />
              </div>
            )
          }}
        </Query>
      </div>
    )
  }
}

Dashboard.propTypes = {
  classes: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired
}

export default withStyles(styles)(Dashboard)
