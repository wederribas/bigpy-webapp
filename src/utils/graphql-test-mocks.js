import { getCompaniesQuery } from '../components/Search/Search'

export const mocks = [
  {
    request: {
      query: getCompaniesQuery
    },
    result: {
      data: {
        allCompaniesNames: {
          edges: [
            { node: { Id: 'Company 0' } },
            { node: { Id: 'Company 1' } },
            { node: { Id: 'Company 2' } }
          ]
        }
      }
    }
  }
]
