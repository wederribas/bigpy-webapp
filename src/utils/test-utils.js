import React from 'react'
import { render } from 'react-testing-library'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  }
})

const customRender = node => {
  return render(<MuiThemeProvider theme={theme}>{node}</MuiThemeProvider>)
}

export * from 'react-testing-library'

export { customRender as render }
