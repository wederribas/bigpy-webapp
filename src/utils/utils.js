// @flow

/**
 * Deburr special Latin letters to basic Latin letters and removing
 * combining diacritical marks.
 */
export function deburr(string: string): string {
  return string.normalize('NFD').replace(/[\u0300-\u036f]/g, '')
}
