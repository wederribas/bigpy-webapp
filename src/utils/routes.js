import Search from '../components/Search/Search'
import Top10 from '../components/Top10/Top10'
import Insights from '../components/Insights/Insights'
import About from '../components/About/About'
import Dashboard from '../components/Dashboard/Dashboard'

export const routes = [
  {
    path: '/',
    component: Search,
    exact: true
  },
  {
    path: '/dashboard/:companyName',
    component: Dashboard
  },
  {
    path: '/top10',
    component: Top10
  },
  {
    path: '/insights',
    component: Insights
  },
  {
    path: '/about',
    component: About
  }
]
