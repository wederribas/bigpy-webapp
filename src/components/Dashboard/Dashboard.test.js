import React from 'react'
import { MockedProvider } from 'react-apollo/test-utils'
import wait from 'waait'
import { render } from '../../utils/test-utils'
import Dashboard, { DASHBOARD_QUERY } from './Dashboard'

const queryData = {
  companyName: 'Company 0',
  totalReportsCount: 1000
}

const dashboardMock = {
  request: {
    query: DASHBOARD_QUERY,
    variables: {
      companyName: queryData.companyName
    }
  },
  result: {
    data: {
      totalCompanyReports: queryData.totalReportsCount
    }
  }
}

test('<Dashboard /> component renders', async () => {
  const match = { params: { companyName: 'Company 0' } }

  const { container, getByTestId, rerender } = render(
    <MockedProvider mocks={[dashboardMock]} addTypename={false}>
      <Dashboard match={match} />
    </MockedProvider>
  )

  expect(container.firstChild).toMatchSnapshot()

  await wait(200)

  expect(getByTestId('company-name')).toHaveTextContent(queryData.companyName)
  expect(getByTestId('reports-count')).toHaveTextContent(
    queryData.totalReportsCount
  )

  // Re-render the components without the query variables
  // in order to cause a GraphQL error
  dashboardMock.request.variables = {}
  rerender(
    <MockedProvider mocks={[dashboardMock]} addTypename={false}>
      <Dashboard match={match} />
    </MockedProvider>
  )

  await wait(200)

  expect(getByTestId('graphql-error')).toBeTruthy()
})
