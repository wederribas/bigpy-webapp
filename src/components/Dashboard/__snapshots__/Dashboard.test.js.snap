// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> component renders 1`] = `
<div>
  <div
    class="Dashboard-loader-2"
  >
    <div
      class="bouncing-loader"
    >
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
